assets/files/etc/kubernetes/manifests/haproxy.yaml

```diff
@@ -68,15 +68,33 @@ spec:
     imagePullPolicy: IfNotPresent
   containers:
   - name: haproxy
-    image: docker.io/library/haproxy:latest
-    args:
-    - "-W"
-    - "-db"
-    - "-S"
-    - "/var/run/haproxy/haproxy-master.sock,level,admin"
-    - "-f"
-    - "/etc/haproxy/haproxy.cfg"
+    image: openshift/origin-haproxy-router:latest
+    command:
+    - "/bin/bash"
+    - "-c"
+    - |
+      #/bin/bash
+      set -ex
+      function reload_haproxy
+      {
+        old_pids=$(pidof haproxy)
+        if [ -n "$old_pids" ]; then
+            /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid -x /var/lib/haproxy/run/haproxy.sock -sf $old_pids &
+        else
+            /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid &
+        fi
+      }
+      /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid &
+      while true; do
+          sleep 10
+          if  grep -q RELOAD /haproxy_reload/msgq; then
+              reload_haproxy
+              echo "" > /haproxy_reload/msgq
+          fi
```

Comment:
I think there’s a race condition here where a config change could get missed. A second config change could come in after a reload but before the msgq file is cleared. That change would be missed.

I know it seems incredibly unlikely. I can think of two options. 

1 - use locking on msgq as the shared resource. 

2 - treat msgq as a queue. Append a line for each reload request and only remove a line when reloading.